### PR TITLE
Update to Kotlin 1.6.20 and update other dependencies

### PR DESCRIPTION
**Pull request by NoComment1105**
No code changes, just dependency updates

### CODE REVIEW

**Comment by Lukellmann** [comment]
Where there some compile warnings for the `-Xopt-in` flag? I remember seeing those in my projects with 1.6.20 pre-release versions.

**Comment by NoComment1105** [comment]
There were. I can fix them if you'd like. `-Xopt-in` has been deprecated for `-opt-in`

**Comment by Lukellmann** [comment]
> There were. I can fix them if you'd like. `-Xopt-in` has been deprecated for `-opt-in`

would be nice 👍

**Comment by NoComment1105** [comment]
All sorted

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.4-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.4.2-all.zip
```
Comment (Lukellmann) [inline gradle/wrapper/gradle-wrapper.properties:3]: Can you also update `gradleVersion` in the root `build.gradle.kts`.
This is used by the `wrapper` task which is the [recommended way to upgrade gradle](https://docs.gradle.org/current/userguide/gradle_wrapper.html#sec:upgrading_wrapper).

Comment (NoComment1105) [reply gradle/wrapper/gradle-wrapper.properties:3]: I already have :>

Comment (Lukellmann) [reply gradle/wrapper/gradle-wrapper.properties:3]: oops, I'm blind :D

**Review by Lukellmann — APPROVED** [review]

**Comment by Lukellmann** [comment]
ktor 2.0.0 is right around the corner (there is an extra PR for that), but maybe we should update to already released 1.6.8 in this one?

File: build.gradle.kts [hunk 2]
```diff
@@ -4,14 +4,14 @@ import org.gradle.api.tasks.wrapper.Wrapper.DistributionType.ALL
 plugins {
     kotlin("jvm")
     kotlin("plugin.serialization")
-    id("org.jetbrains.kotlinx.binary-compatibility-validator") version "0.6.0"
+    id("org.jetbrains.kotlinx.binary-compatibility-validator") version "0.8.0"
     id("org.jetbrains.dokka")
 
-    id("org.ajoberstar.git-publish") version "2.1.3"
+    id("org.ajoberstar.git-publish") version "3.0.1"
```
Comment (NyCodeGHG) [inline build.gradle.kts:10]:
```suggestion
    id("org.ajoberstar.git-publish") version "4.1.0"
```

**Review by NyCodeGHG — COMMENTED** [review]
git-publish's latest version is 4.1.0, but it's not shown on plugins.gradle.org, because they now publish to maven central (https://github.com/ajoberstar/gradle-git-publish#newest-versions-are-on-maven-central)

**Comment by NyCodeGHG** [comment]
building with java 11+ would be an option, while still compiling on jvm target 8

**Comment by NoComment1105** [comment]
> ktor 2.0.0 is right around the corner (there is an extra PR for that), but maybe we should update to already released 1.6.8 in this one?

That can be done, sure.

**Comment by NoComment1105** [comment]
Done! There was no code changing required.

**Review by Lukellmann — APPROVED** [review]

**Pull request merged**